src/main/java/com/commercetools/sync/commons/helpers/GraphQlRequest.java

```diff
@@ -0,0 +1,129 @@
+     * @param endpoint - a string representing the name of the resource endpoint.
+     */
+    public GraphQlRequest(@Nonnull final Set<String> keysToSearch, @Nonnull final GraphQlQueryEndpoint endpoint) {
+
+        this.keysToSearch = requireNonNull(keysToSearch);
+        this.endpoint = endpoint;
+    }
+
+    /**
+     * This method adds a predicate string to the request.
+     *
+     * @param predicate - a string representing a query predicate.
+     * @return - an instance of this class.
+     */
+    @Nonnull
+    public GraphQlRequest withPredicate(final String predicate) {
+
+        this.queryPredicate = predicate;
+        return this;
+    }
+
+    /**
+     * This method adds a limit to the request.
+     *
+     * @param limit - a number representing the query limit parameter
+     * @return - an instance of this class
+     */
+    @Nonnull
+    public GraphQlRequest withLimit(final long limit) {
+
+        this.limit = limit;
+        return this;
+    }
+
+    @Nullable
+    @Override
+    public GraphQlResult deserialize(final HttpResponse httpResponse) {
+
+        final JsonNode rootJsonNode = SphereJsonUtils.parse(httpResponse.getResponseBody());
+        if (rootJsonNode.isNull()) {
+            return null;
+        }
+        JsonNode result = rootJsonNode.get("data").get(endpoint.getName());
+        return SphereJsonUtils.readObject(result, GraphQlResult.class);
+    }
+
+    @Override
+    public HttpRequestIntent httpRequestIntent() {
+
+        final String body = format("{\"query\": \"{%s}\"}", buildQueryString());
+        return HttpRequestIntent.of(HttpMethod.POST, "/graphql", body);
+    }
+
+    /**
+     * This method builds a string matching the required format to query a set of ids matching given keys of a
+     * resource using the CTP graphql API
+     *
+     * @return a string representing a graphql query
+     */
+    @Nonnull
+    String buildQueryString() {
+
+        return format(
+            "%s(limit: %d, where: \\\"%s\\\", sort: [\\\"id asc\\\"]) { results { id key } }",
+            this.endpoint.getName(), this.limit, createWhereQuery(keysToSearch));
+    }
+
+    @Nonnull
+    private String createWhereQuery(@Nonnull final Set<String> keys) {
+        // The where in the graphql query should look like this in the end =>  `where: "key in (\"key1\",
+        // \"key2\")"`
+        // So we need an escaping backslash before the quote. So to add this:
+        // We need 1 backslash (2 in java) to escape the quote in the graphql query.
+        // We need 2 backslashes (4 in java) to escape the backslash in the JSON payload string.
+        // We need 1 extra backslash to escape the quote in the java string
+        // hence: 7 backslashes:
+        final String backslashQuote = "\\\\\\\"";
+        final String commaSeparatedKeys =
+            keys.stream()
+                .collect(
+                    joining(
+                        format("%s, %s", backslashQuote, backslashQuote),
```

Comment:
That's a good point, maybe it should be thrown as illegal arguments, also keys could be null 🤔 

The probability seems low because maybe the service is doing those checks, but a key in the set could be empty or null, anyways to make sure it could be validated in the constructor. We could also check how the rest calls or SDK were handling those cases.

Reply:
@lojzatran empty `keys` Set is possible, but no problem for the query - it will return empty results.
haven't tried with null or empty values in Set yet.

Reply:
@lojzatran @ahmetoz Its no problem for the SDK to handle null and empty values, it responds with empty result. Constructor has a `requireNonNull` check. What about filtering null and empty keys in this method:
`final String commaSeparatedKeys =
            keys.stream()
                .filter(key -> isBlank(key))
                .collect(
                    joining(
                        format("%s, %s", backslashQuote, backslashQuote),
                        backslashQuote,
                        backslashQuote));`

Reply:
👍 It makes sense to filter as you suggest.

Reply:
2e9572f